### PR TITLE
Support optional `kpt_oks_sigmas` for pose evaluation via dataset YAML

### LINKED ISSUE #24349: Feature Request: Allow dynamic OKS sigma values for pose evaluation metrics via dataset YAML
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Following up on @glenn-jocher's comment in #23843, I am proposing a solution to start cleaning up the hardcoded `OKS_SIGMA` logic, starting with the evaluation metrics.

Currently, `init_metrics` in `ultralytics/models/yolo/pose/val.py` defaults to COCO sigmas or uniform sigmas. I propose allowing users to pass these dynamically via the dataset YAML.

The updated logic in `val.py` would look like this:

```python
# Check if custom OKS sigmas are provided in the dataset YAML
if 'kpt_oks_sigmas' in self.data:
    self.sigma = np.array(self.data['kpt_oks_sigmas'])
else:
    self.sigma = OKS_SIGMA if is_pose else np.ones(nkpt) / nkpt
```

### Use case

```markdown
When training and evaluating pose models on custom datasets (e.g., non-human subjects, animals, or industrial objects), the default COCO-based human sigmas are not applicable. 

As seen in previous community questions (e.g., #16141, #22527, and #23843), users frequently struggle with applying custom sigma values, often resorting to manually monkeypatching the source code. Allowing these values to be passed dynamically via the dataset YAML provides much-needed flexibility without breaking existing workflows.
```

### Additional

**Note regarding Training Loss & Tests:** As Glenn mentioned in #23843, `v8PoseLoss` also needs this cleanup. To keep PRs small and manageable as requested, I intentionally kept this initial PR focused solely on `val.py` for evaluation metrics. 

I already have the code ready on my fork and I am ready to submit the PR as soon as you give the green light.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by cosmo-gb**
Closes #24349 

### Description
Implements the changes requested by @glenn-jocher in the issue linked above:
- Added `eval-only` support for dynamic OKS sigmas via dataset YAML (`kpt_oks_sigmas`).
- Added validation to raise a `ValueError` if the length of the provided sigmas does not match `kpt_shape[0]`.
- Added a focused unit test for this behavior in `tests/test_pose_custom_oks.py`.
- Added a brief docs note in `docs/en/guides/model-yaml-config.md`.

### CLA
I have read the CLA Document and I sign the CLA.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds optional `kpt_oks_sigmas` support for pose evaluation via dataset YAML, enabling custom OKS sigma values with validation and test coverage. 🎯

### 📊 Key Changes
- Added support in `PoseValidator.init_metrics()` to read optional `kpt_oks_sigmas` from dataset YAML for pose evaluation.
- Added validation to ensure the provided `kpt_oks_sigmas` length exactly matches the keypoint count defined by `kpt_shape`, raising a clear `ValueError` when mismatched.
- Preserved existing default behavior by continuing to use built-in OKS sigmas for standard `[17, 3]` pose datasets and uniform fallback values for other keypoint layouts when custom sigmas are not provided.
- Added a new test file `tests/test_pose_custom_oks.py` covering both valid custom sigma loading and invalid length handling. ✅
- Updated documentation in `docs/en/guides/model-yaml-config.md` to describe the new optional dataset YAML parameter and its expected format. 📘

### 🎯 Purpose & Impact
- Gives pose users more flexibility to evaluate custom keypoint datasets with dataset-specific OKS sigma settings.
- Improves robustness by validating configuration errors early instead of silently using incorrect settings.
- Enhances usability through documentation and tests, making the new behavior easier to adopt and maintain.
- Benefits pose workflows in the Ultralytics package without changing default behavior for existing datasets. 🚀

---

## Maintainer update

Deleted: `tests/test_pose_custom_oks.py` (bundled test removed per repo preference, behavior verified empirically) and the docs note in `guides/model-yaml-config.md` (wrong page — `kpt_oks_sigmas` is a dataset YAML key, not a model YAML key), relocated to `datasets/pose/index.md` beside the optional `flip_idx` key. The validation was compacted to a single `self.data.get()` branch reusing the existing `OKS_SIGMA` fallback and the `kpt_iou` consumer — no new helper added.

Verified empirically with `yolo11n-pose.pt` on `coco8-pose`: custom `kpt_oks_sigmas: [0.1] * 17` changes pose mAP50-95 (0.3094 → 0.465, confirming the sigmas flow into `kpt_iou`), omitting the key reproduces the default COCO-sigma metrics exactly, and wrong length (10 of 17) or non-positive values raise a clear `ValueError` at `init_metrics`.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cosmo-gb, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an engineer will assist you soon. Please review the checklist below to help ensure smooth integration of your changes 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/models/yolo/pose/val.py [hunk 1]
```diff
@@ -102,7 +102,14 @@ def init_metrics(self, model: torch.nn.Module) -> None:
         self.kpt_shape = self.data["kpt_shape"]
         is_pose = self.kpt_shape == [17, 3]
         nkpt = self.kpt_shape[0]
-        self.sigma = OKS_SIGMA if is_pose else np.ones(nkpt) / nkpt
+        # Check if custom OKS sigmas are provided in the dataset YAML
+        if 'kpt_oks_sigmas' in self.data:
+            sigmas = np.array(self.data['kpt_oks_sigmas'])
```
Comment (UltralyticsAssistant) [inline ultralytics/models/yolo/pose/val.py]: 💡 **MEDIUM**: This only validates the outer length. Malformed YAML such as nested lists can still pass here with a non-1D array, and zero/negative sigma values will make OKS invalid or trigger divide-by-zero downstream. Please coerce to a 1-D float array first and reject any non-positive entries so bad configs fail with a clear `ValueError` at parse time.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Useful feature and the default/custom fallback logic looks good overall. The main gap is input validation for `kpt_oks_sigmas`: it should enforce a 1-D numeric array with strictly positive values to avoid incorrect metrics or cryptic runtime failures.

💬 Posted 1 inline comment

**Comment by cosmo-gb** [comment]
I have read the CLA Document and I sign the CLA

File: ultralytics/models/yolo/pose/val.py (hunk 1 above)
Comment (cosmo-gb) [reply ultralytics/models/yolo/pose/val.py]: Done in the latest commit !

**Comment by cosmo-gb** [comment]
Hi @glenn-jocher, just a friendly ping on this PR whenever you have a moment. Just let me know if you need any tweaks before merging! Thanks.

**Comment by glenn-jocher** [comment]
Thanks for the ping and for keeping this focused with test coverage and a docs note; the `kpt_oks_sigmas` dataset-YAML approach in `PoseValidator.init_metrics()` looks reasonable, and if any changes are needed during review I’ll leave them here.

**Review by glenn-jocher — APPROVED** [review]
LGTM at exact head 6ce53b8. PoseValidator.init_metrics is the narrowest evaluation owner, the default path is unchanged, and the dataset documentation is colocated with other pose dataset keys. Delete/replace review: standalone test and misplaced docs were removed; no added condition suppresses a symptom—the validation rejects invalid configuration at its consumer.
